### PR TITLE
Add help component

### PR DESCRIPTION
**Pull request by christianmemije**
### What's included?

Adds a help component 

### Future work?

Be able to parse items from a JSON file.
See all TODOs in PR

#### Test Steps

- [ ] `npm i`
- [ ] `npm run serve:experimental`

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

##### Screenshots or link to StackBlitz/Plunker

help_component_demo

### CODE REVIEW

**Comment by stevenov7** [comment]
I tried entering a bad url:
Screen Shot 2019-06-12 at 1 42 02 PM

and got this screen:
Screen Shot 2019-06-12 at 1 42 38 PM

We should probably display some visible message to let the user know something went wrong.

**Comment by stevenov7** [comment]
I tried passing in an empty array. 
Screen Shot 2019-06-12 at 1 46 51 PM

and got this screen
Screen Shot 2019-06-12 at 1 47 45 PM
we should use some form of the Covalent empty state pattern in this case:
https://teradata.github.io/covalent/#/design-patterns/empty-states

**Comment by stevenov7** [comment]
When i pass in a title that is semi long:
Screen Shot 2019-06-12 at 1 56 01 PM

the title should be ellipsis and have a tooltip for when it overflows:
Screen Shot 2019-06-12 at 1 55 02 PM

**Comment by stevenov7** [comment]
Issue on Safari and Firefox.
Issue not on Chrome.
When a user scroll in the dialog on Safari and Firefox the whole page scrolls when hitting the bottom of the list.
rec-1

**Comment by stevenov7** [comment]
Testing the `anchor` feature with this config:
Screen Shot 2019-06-12 at 3 54 27 PM
it doesn't jump to `Usage` it jumps to `Properties`

File: src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    public dialogRef: MatDialogRef<DraggableHelpWindowDialogComponent>,
+  ) {}
+
+  handleClosed(): void {
+    this.dialogRef.close();
+  }
+
+  handleMinimized(): void {
+    // TODO: not working
```
Comment (jeremysmartt) [inline src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.component.ts]: is this todo comment accurate?

File: src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    this.dialogRef.updatePosition({bottom: '0', right: '0'});
+  }
+  handleMaximized(): void {
+    // TODO: not working
```
Comment (jeremysmartt) [inline src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.component.ts]: is this todo comment accurate?

File: src/platform/experimental/help/help.module.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    MatButtonModule,
+    MatTooltipModule,
+    MatListModule,
+    MatIconModule,
+    MatDialogModule,
+    MatProgressBarModule,
+    DragDropModule,
+    MatToolbarModule,
+
+    // TODO: not used, but should be able to just use normal md comp
```
Comment (jeremysmartt) [inline src/platform/experimental/help/help.module.ts]: is this todo comment accurate?

File: src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.service.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import { Overlay, NoopScrollStrategy } from '@angular/cdk/overlay';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DraggableHelpWindowDialogService {
+  scrollStrategy: NoopScrollStrategy;
+
+  constructor(private _dialog: MatDialog, overlay: Overlay) {
+    this.scrollStrategy = overlay.scrollStrategies.noop();
```
Comment (emoralesb05) [inline src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.service.ts]: Why did you have to change the scroll strat?

File: src/platform/experimental/help/help.module.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    HelpWindowComponent,
+    HelpWindowToolbarComponent,
+    DraggableHelpWindowDialogComponent,
+    TdMarkdownLoaderComponent,
+  ],
+  exports: [
+    HelpComponent,
+    HelpWindowComponent,
+    DraggableHelpWindowDialogComponent,
+    TdMarkdownLoaderComponent,
```
Comment (emoralesb05) [inline src/platform/experimental/help/help.module.ts]: Should this be exported here ? or in the `@covalent/markdown` module?

Comment (emoralesb05) [reply src/platform/experimental/help/help.module.ts]: Ah, i see you added it as part of this module.. wondering if this should be part of `markdown` module

File: src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.service.ts (hunk 4 above)
Comment (christianmemije) [reply src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.service.ts]: Damnit, I forgot what it fixed. >.<

File: src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.component.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { Component, OnInit, Inject } from '@angular/core';
```
Comment (richavyas) [inline src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.component.ts]: OnInit is not used anywhere.

File: src/platform/experimental/help/help/help.component.html [hunk 7]
```diff
@@ -0,0 +1,86 @@
+<ng-container *ngIf="!showEmptyState">
```
Comment (richavyas) [inline src/platform/experimental/help/help/help.component.html]: why do we have `help` folder inside `help` folder? Can we not keep those component files on help root level?

File: src/platform/experimental/help/help/help.component.ts [hunk 8]
```diff
@@ -0,0 +1,203 @@
+import { Component, Input, OnChanges, ElementRef, ViewChild, OnDestroy, SimpleChanges } from '@angular/core';
```
Comment (richavyas) [inline src/platform/experimental/help/help/help.component.ts]: ViewChild isn't used anywhere.

File: src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.service.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import { Injectable, TemplateRef } from '@angular/core';
```
Comment (richavyas) [inline src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.service.ts]: `TemplateRef` not used anywhere.

File: src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.component.ts (hunk 1 above)
Comment (christianmemije) [reply src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.component.ts]: Removed minimize/maximize code for now.

File: src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.component.ts (hunk 2 above)
Comment (christianmemije) [reply src/platform/experimental/help/draggable-help-window-dialog/draggable-help-window-dialog.component.ts]: Removed minimize/maximize code for now.

File: src/platform/experimental/help/help/help.component.html (hunk 7 above)
Comment (christianmemije) [reply src/platform/experimental/help/help/help.component.html]: Updated.

**Pull request merged**